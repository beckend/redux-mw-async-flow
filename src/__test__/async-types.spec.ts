/* tslint:disable: chai-vague-errors */
import {
  getAsyncTypeConstants,
  generateAsyncAction
} from '../async-types';

describe('async-types', () => {
  it('getAsyncTypeConstants', () => {
    expect(getAsyncTypeConstants({
      types: {
        REQUEST: '_REQUEST___',
        PENDING: '_PENDING___',
        FULFILLED: '_FULFILLED___',
        REJECTED: '_REJECTED___',
        ABORTED: '_ABORTED___',
        END: '_END___'
      }
    }))
      .toMatchSnapshot();
  });

  it('generateAsyncAction with types', () => {
    expect(generateAsyncAction({
      actionName: 'namespaceAnother/MYACTION',
      types: {
        REQUEST: '_REQUEST___',
        PENDING: '_PENDING___',
        FULFILLED: '_FULFILLED___',
        REJECTED: '_REJECTED___',
        ABORTED: '_ABORTED___',
        END: '_END___'
      }
    }))
      .toMatchSnapshot();
  });
  it('generateAsyncAction with defaults', () => {
    expect(generateAsyncAction({
      actionName: 'namespace/MYACTION'
    }))
      .toMatchSnapshot();
  });
});
