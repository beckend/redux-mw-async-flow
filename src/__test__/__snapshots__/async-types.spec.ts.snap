exports[`async-types generateAsyncAction with defaults 1`] = `
Object {
  "namespace/MYACTION_ABORTED": "namespace/MYACTION_ABORTED",
  "namespace/MYACTION_END": "namespace/MYACTION_END",
  "namespace/MYACTION_FULFILLED": "namespace/MYACTION_FULFILLED",
  "namespace/MYACTION_PENDING": "namespace/MYACTION_PENDING",
  "namespace/MYACTION_REJECTED": "namespace/MYACTION_REJECTED",
  "namespace/MYACTION_REQUEST": "namespace/MYACTION_REQUEST",
}
`;

exports[`async-types generateAsyncAction with types 1`] = `
Object {
  "namespaceAnother/MYACTION_ABORTED___": "namespaceAnother/MYACTION_ABORTED___",
  "namespaceAnother/MYACTION_END___": "namespaceAnother/MYACTION_END___",
  "namespaceAnother/MYACTION_FULFILLED___": "namespaceAnother/MYACTION_FULFILLED___",
  "namespaceAnother/MYACTION_PENDING___": "namespaceAnother/MYACTION_PENDING___",
  "namespaceAnother/MYACTION_REJECTED___": "namespaceAnother/MYACTION_REJECTED___",
  "namespaceAnother/MYACTION_REQUEST___": "namespaceAnother/MYACTION_REQUEST___",
}
`;

exports[`async-types getAsyncTypeConstants 1`] = `
Object {
  "ABORTED": "_ABORTED___",
  "END": "_END___",
  "FULFILLED": "_FULFILLED___",
  "PENDING": "_PENDING___",
  "REJECTED": "_REJECTED___",
  "REQUEST": "_REQUEST___",
}
`;
